unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarShape.cs

```diff
@@ -195,6 +195,17 @@ public override IEnumerator ApplyChanges(BaseModel newModel)
                     playerName.SetName(model.name);
                     playerName.Show(true);
                 }
+
+                // NOTE (Kinerius): Since users can spawn avatar shapes with zero scale,
+                // we avoid the glitches generated by it by moving this object elsewhere with a normal scale (#2997)
+                var thisTransform = transform;
+                Transform prevParent = thisTransform.parent;
+                Vector3 prevScale = thisTransform.localScale;
+                Vector3 prevPos = thisTransform.localPosition;
+                thisTransform.SetParent(null);
+                thisTransform.localPosition = Vector3.one * 9999;
```

Comment:
cant we disable the gameObject instead of doing this?

Reply:
Disabling would do nothing because it would still be a child of an object hierarchy of zero scale

Reply:
so what if you unparent and disable the object?